src/java.base/share/classes/java/io/InputStream.java

```diff
@@ -594,20 +595,19 @@ public long skip(long n) throws IOException {
      * @see        java.io.InputStream#skip(long)
      */
     public void skipNBytes(long n) throws IOException {
-        if (n > 0) {
+        while (n > 0) {
             long ns = skip(n);
-            if (ns >= 0 && ns < n) { // skipped too few bytes
+            if (ns > 0 && ns <= n) {
                 // adjust number to skip
                 n -= ns;
-                // read until requested number skipped or EOS reached
-                while (n > 0 && read() != -1) {
-                    n--;
-                }
-                // if not enough skipped, then EOFE
-                if (n != 0) {
+            } else if (ns == 0) { // no bytes skipped
+                // read one byte to check for EOS
+                if (read() < 0) {
```

Comment:
Wouldn't it be better to compare return value with -1 (as it was before)?
Since the documentation of read method says:
_Returns:
    the next byte of data, or **-1 if the end of the stream is reached**._

Reply:
Indeed that would match the letter of the specification of `read()` but I don't think it would make any practical difference.